spec/support/matchers/have_actions.rb

```diff
@@ -57,18 +57,38 @@ def unless_option_for(action)
   action.instance_variable_get(:@unless)
 end
 
-def symbol_from_string(string)
-  if string.include?('||')
-    string.split('||').map { |str| str.split('==')[1].strip.tr("'", '').to_sym }
-  else
-    string.split('==')[1].strip.tr("'", '').to_sym
-  end
+def parsed_only_action(action)
+  only_option = if_option_for(action)[0]
+
+  "#{only_option.class}OptionParser".constantize.new(only_option).parse
+end
+
+def parsed_except_action(action)
+  except_option = unless_option_for(action)[0]
+
+  "#{except_option.class}OptionParser".constantize.new(except_option).parse
 end
 
-def symbolized_only_action(action)
-  symbol_from_string(if_option_for(action)[0])
+class ProcOptionParser
+  def initialize(option)
+    @option = option
+  end
+
+  def parse
+    @option
+  end
 end
 
-def symbolized_except_action(action)
-  symbol_from_string(unless_option_for(action)[0])
+class StringOptionParser
+  def initialize(option)
+    @option = option
+  end
+
+  def parse
+    if @option.include?('||')
+      @option.split('||').map { |str| str.split('==')[1].strip.tr("'", '').to_sym }
+    else
+      @option.split('==')[1].strip.tr("'", '').to_sym
```

Comment:
Good call. I made the change in my second commit. Let me know if the method names are not clear enough.